tests/dtypes_test.py

```diff
@@ -138,6 +143,23 @@ def testPromoteDtypes(self):
           self.assertEqual(np.promote_types(t1, t2),
                            dtypes.promote_types(t1, t2))
 
+  # TODO(jakevdp): fix these breakages and enable this test.
+  @skipIf(True, "Binary promotion is not yet invariant under jit in some cases.")
+  @parameterized.named_parameters(
+    {"testcase_name": "xtype={}_ytype={}_xfun={}_yfun={}".format(
+      xtype.__name__, ytype.__name__, xfun.__name__, yfun.__name__),
+     "xtype": xtype, "ytype": ytype, "xfun": xfun, "yfun": yfun}
+    for xtype, ytype in itertools.product(
+      [int, float, jnp.int16, jnp.int32, jnp.float16, jnp.float32], repeat=2)
+    for xfun, yfun in itertools.product(
+      [identity, abs, jnp.array], repeat=2)
+    )
+  def testBinaryPromotionJitInvariance(self, xtype, ytype, xfun, yfun):
+    """Test jit invariance of simple binary promotion rules with and without weak types."""
+    f = lambda x, y: xfun(x) + yfun(y)
+    args_maker = lambda: [xtype(1), ytype(1)]
+    self._CompileAndCheck(f, args_maker, check_dtypes=True)
```

Comment:
This test is not strictly necessary for this PR, but it is a good TDD case for the next step in implementing weak types, so I thought it would be useful to include.